addon/nodes/paragraphWithConfig.ts

```diff
@@ -20,6 +20,9 @@ BLOCK_SELECTOR = `:not(${BLOCK_SELECTOR.substring(
 )})`;
 
 const BASE_PARAGRAPH_TYPE = 'paragraph';
+
+export const DERIVED_FROM_PARAGRAPH = 'paragraph';
+
```

Comment:
That's true, but I'd see that as a feature. If you don't add a special paragraph to `paragraphGroup`, you pretty much choose you want to customize the logic yourself.
And with your code, it would still work: all it blocks is the special case of allowing splitting up a paragraph.

I do get your solution though. It gives more freedom. But I think I lean more to keeping it simple for now and using the groups, and avoid adding a very specific nodespec for this specific case. If this difference is ever needed, that case can then be looked at on its own and solved in a more direct way.
E.g. this "functionality" could be solved more generally by creating `splittable` nodespec that you add to nodes that can be split by typical basic blocks (like tables and such). Adding such a nodespec would make sense if we want this logic for more than just our paragraphs.

Hopefully that makes sense. No correct answer here though, just my opinion :shrug: